iothub/device/src/Microsoft.Azure.Devices.Client.csproj

```diff
@@ -52,13 +52,15 @@
   <ItemGroup>
     <PackageReference Include="MQTTnet" Version="4.1.2.350" />
     <PackageReference Include="Microsoft.Azure.Amqp" Version="2.5.12" />
+    <PackageReference Include="System.Text.Json" Version="5.0.0" />
```

Comment:
Trying out a reference to stj at v5 (first to include `[JsonInclude]` to solve the deserialization problem for internal set properties.